src/instructlab/config/show.py

```diff
@@ -14,15 +16,53 @@
 yaml.indent(mapping=2, sequence=4, offset=2)
 
 
+def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
+    """Retrieve nested value using dot-separated path."""
+    keys = path.split(".")
+    for key in keys:
+        data_dict = data_dict[key]
+    return data_dict
+
+
 @click.command()
 @click.pass_context
+@click.option(
+    "--json",
```

Comment:
This option is complicated by having logging output to stdout.  I think we will need to address moving logging to a separate file before we can support json output consistently.  See https://github.com/instructlab/instructlab/pull/2470 for additional conversation on the topic.